### PR TITLE
Remove NaNs from data before components calculation

### PR DESCRIPTION
**Pull request by oesteban**
Closes poldracklab/fmriprep#776

### CODE REVIEW

**Comment by effigies** [comment]
Are you sure there are NaNs, and it's not just singular?

**Comment by oesteban** [comment]
Not sure, but svd won't take NaNs anyways.

File: nipype/algorithms/confounds.py [hunk 1]
```diff
@@ -1067,7 +1067,7 @@ def compute_noise_components(imgseries, mask_images, num_components,
 
         # "The covariance matrix C = MMT was constructed and decomposed into its
         # principal components using a singular value decomposition."
-        u, _, _ = linalg.svd(M, full_matrices=False)
+        u, _, _ = linalg.svd(M[~np.isnan(M)], full_matrices=False)
```
Comment (satra) [inline nipype/algorithms/confounds.py]: i don't see how anything in `M` would be a `nan` given line 1047 and the `_compute_tSTD` function which should return a 1 if the variance is 0.

is there a particular use case we can look at where you found this?

Comment (oesteban) [reply nipype/algorithms/confounds.py]: I'm going to look closer into this. You're right. And this change would reshape `M` so it basically breaks the interface.

**Comment by oesteban** [comment]
Confirmed this problem is not within Nipype's realm

**Pull request closed without merging**